### PR TITLE
Adds config inclusion system

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl: 
config: Added "$include" directives to config files. These are recursive. Only config.txt will be default loaded if they are specified inside it
/:cl:

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -3,8 +3,47 @@
 GLOBAL_VAR(security_mode)
 GLOBAL_VAR(restart_counter)
 GLOBAL_PROTECT(security_mode)
+/world
+	loop_checks = FALSE
 
 /world/New()
+	var/list/all_mobs = list()
```
Comment (RemieRichards) [inline code/game/world.dm]: good job

Comment (Cyberboss) [reply code/game/world.dm]: Fuck

File: code/controllers/configuration/entries/general.dm [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+/datum/config_entry/flag/log_emote	// log emotes
+
+/datum/config_entry/flag/log_adminchat	// log admin chat messages
+	protection = CONFIG_ENTRY_LOCKED
+
+/datum/config_entry/flag/log_pda	// log pda messages
+
+/datum/config_entry/flag/log_twitter	// log certain expliotable parrots and other such fun things in a JSON file of twitter valid phrases.
+
+/datum/config_entry/flag/log_world_topic)	// log all world.Topic( calls
```
Comment (kevinz000) [inline code/controllers/configuration/entries/general.dm]: )

File: code/controllers/configuration/entries/general.dm [hunk 3]
```diff
@@ -0,0 +1,385 @@
+/datum/config_entry/flag/autoadmin  // if autoadmin is enabled
+	protection = CONFIG_ENTRY_LOCKED
+
+/datum/config_entry/string/autoadmin_rank	// the rank for autoadmins
+	value = "Game Master"
+	protection = CONFIG_ENTRY_LOCKED
+
+/datum/config_entry/string/servername)	// server name (the name of the game window
```
Comment (kevinz000) [inline code/controllers/configuration/entries/general.dm]: )

File: code/controllers/configuration/entries/general.dm [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+
+/datum/config_entry/string/autoadmin_rank	// the rank for autoadmins
+	value = "Game Master"
+	protection = CONFIG_ENTRY_LOCKED
+
+/datum/config_entry/string/servername)	// server name (the name of the game window
+
+/datum/config_entry/string/serversqlname	// short form server name used for the DB
+
+/datum/config_entry/string/stationname)	// station name (the name of the station in-game
```
Comment (kevinz000) [inline code/controllers/configuration/entries/general.dm]: )

File: code/controllers/configuration/entries/general.dm [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+/datum/config_entry/flag/log_manifest	// log crew manifest to seperate file
+
+/datum/config_entry/flag/allow_admin_ooccolor	// Allows admins with relevant permissions to have their own ooc colour
+
+/datum/config_entry/flag/allow_vote_restart	// allow votes to restart
+
+/datum/config_entry/flag/allow_vote_mode	// allow votes to change mode
+
+/datum/config_entry/number/vote_delay)	// minimum time between voting sessions (deciseconds, 10 minute default
```
Comment (kevinz000) [inline code/controllers/configuration/entries/general.dm]: ) and one below too

File: code/controllers/configuration/entries/general.dm [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+
+/datum/config_entry/string/panic_address/ValidateAndSet(str_val)
+	return str_val != "byond://address:port" && ..()
+
+/datum/config_entry/string/invoke_youtubedl
+	protection = CONFIG_ENTRY_LOCKED | CONFIG_ENTRY_HIDDEN
+
+/datum/config_entry/flag/show_irc_name
+
+/datum/config_entry/flag/see_own_notes)	//Can players see their own admin notes (read-only?
```
Comment (kevinz000) [inline code/controllers/configuration/entries/general.dm]: )

File: code/controllers/configuration/entries/general.dm (hunk 2 above)
Comment (kevinz000) [reply code/controllers/configuration/entries/general.dm]: also Topic() :P

**Comment by kevinz000** [comment]
regex whoop whoop

**Comment by Cyberboss** [comment]
I'm honestly surprised the regex didn't work, it was pretty simple

**Comment by optimumtact** [comment]
do you even compile your pr's before pushing them

**Comment by Cyberboss** [comment]
No, because it takes literally 10 minutes to compile

**Comment by ShizCalev** [comment]
@Cyberboss Conflict

**Comment by optimumtact** [comment]
conflicted

**Pull request merged**